run_tests.sh

```diff
@@ -0,0 +1,120 @@
+#!/bin/bash
+
+# Default values
+NETWORK="regtest"
+LOG_LEVEL="none"
+LONG_TESTS=0
+BACKTRACE=0
+
+# Parse command-line arguments
+while getopts "l:b:t:L" opt; do
+  case $opt in
+    l)
+    if [ "$OPTARG" == "debug" ] || [ "$OPTARG" == "info" ] || [ "$OPTARG" == "none" ]; then
+        LOG_LEVEL=$OPTARG
+    else
+        echo "Invalid log level option. Use 'info', 'debug', or 'none'."
+        exit 1
+    fi
+    ;;
+    b)
+      BACKTRACE=$OPTARG
+      ;;
+    t)
+      TEST_NAME="$OPTARG"
+      ;;
+    L)
+      LONG_TESTS=1
+      ;;
+    \?)
+      echo "Invalid option: -$OPTARG" >&2
+      exit 1
+      ;;
+    :)
+      echo "Option -$OPTARG requires an argument." >&2
+      exit 1
+      ;;
+  esac
+done
+
+# Derive the NETWORK value from ~/.bitcoin/bitcoin.conf
+if grep -q '^regtest=1' ~/.bitcoin/bitcoin.conf; then
+    NETWORK="regtest"
+elif grep -q '^signet=1' ~/.bitcoin/bitcoin.conf; then
+    NETWORK="signet"
+else
+    echo "No valid network configuration found in bitcoin.conf."
+    exit 1
+fi
+
+# Create a wallet if it doesn't exist
+bitcoin-cli createwallet test 2>/dev/null
```

Comment:
this might be useless. I believe that people running a bitcoind already have a wallet

Reply:
removing